### DISCUSSION TITLE
ContactStructuralMechanicsApplication - Python coupling

### DISCUSSION THREAD

**Discussion by sunethwarna** [KratosMultiphysics/Kratos#11774] [Q&A]
Hi all,

We have out that, one of the strategies in `ContactStructuralMechanicsApplication` includes `pybind` headers to get `python::Process` type as shown below:

https://github.com/KratosMultiphysics/Kratos/blob/794c315fa9ee91339f44a66c3001f03e8a4dee08/applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/residualbased_newton_raphson_contact_strategy.h#L33-L34

Followings are some use cases of the `pybind` includes in the same file:
https://github.com/KratosMultiphysics/Kratos/blob/794c315fa9ee91339f44a66c3001f03e8a4dee08/applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/residualbased_newton_raphson_contact_strategy.h#L136
https://github.com/KratosMultiphysics/Kratos/blob/794c315fa9ee91339f44a66c3001f03e8a4dee08/applications/ContactStructuralMechanicsApplication/custom_strategies/custom_strategies/residualbased_newton_raphson_contact_strategy.h#L855-L859

Would it be possible to change these from `python::Process` to `Kratos::Process` and decouple `ContactStructuralMechanicsApplication` to `ContactStructuralMechanicsCore` and `ContactStructuralMechanicsApplication` as in other applications?

@loumalouomega ?

**Comment by loumalouomega** [comment]
I don't remember why I did that, but could be...

**Reply by loumalouomega** [reply]
There is already KratosContactStructuralMechanicsCore and KratosContactStructuralMechanicsApplication

**Comment by sunethwarna** [comment]
I can remove it if it is not killing anyone's use cases. :D

**Reply by loumalouomega** [reply]
I can do it myself, because if I did it that way it was because some reason

**Reply by loumalouomega** [reply]
I don't see where is the problem, application is separtated in the standard manner

**Comment by sunethwarna** [comment]
We got a request that, they would like to use `ContactStructuralMechanicsApplication` without python layer (so they are linking against the cpp part only.) At the moment, they have replaced `ProcessesListType` with `Kratos::Process` which works for them. [They don't like to have pybind11 includes in their program which makes sense]. Is this possible? Am I missing something?

**Reply by loumalouomega** [reply]
This should not work a priori, as in fact the actual argument of the utility are raw python objects then reinterpreted. What I would do is to refactopr in order to remove that dependency. Or maybe just using preprocessor to not compile that part when no python compiled. i will try and tell you.

**Comment by loumalouomega** [comment]
Is this sitll an issue?
